server/pkg/database/boltDB.go

```diff
@@ -229,6 +229,44 @@ func (b *boltDB) GetAccount(username string) (models.Account, error) {
 				account = acc
 				return nil
 			}
+
+		}
+
+		return fmt.Errorf("no matching account found")
```

Comment:
Try to keep the error messages to "real" errors. If you check that `account != account{}` in your logic calling this the error handling flow can stay consistent throughout the code base.

Reply:
This is duplicated from your code on (what is now) GetAccountByUsername do you want the methods to act more like they do for GetAccountByID? 
e.g.
```
		a := bucket.Get(itob(userID))

		return json.Unmarshal(a, &account)
```

Reply:
No, how you have it is perfect. Bolt stores one value as a lookup value so you would have to store the account three times. 

It's a bit of a performance hit but I think it should be better with a switch to Dynamo DB for production in the future.